src/main/java/bio/terra/service/snapshotbuilder/query/Query.java

```diff
@@ -151,7 +103,55 @@ public String renderSQL(SqlRenderContext context) {
     return "SELECT " + sql;
   }
 
-  public SourceVariable getPrimaryTable() {
-    return tables.stream().filter(SourceVariable::isPrimary).findFirst().orElseThrow();
+  public Table getPrimaryTable() {
+    return tables.stream().filter(Table::isPrimary).findFirst().orElseThrow();
+  }
+
+  public static class Builder {
+    private List<SelectExpression> select;
+    private List<Table> tables;
+    private FilterVariable where;
+    private List<FieldVariable> groupBy;
+    private List<OrderByVariable> orderBy;
+    private Integer limit;
+
+    public Builder() {
+      // Constructor intentionally left empty to allow for custom initialization via setter methods
+    }
+
+    public Builder select(List<SelectExpression> select) {
+      this.select = select;
+      return this;
+    }
+
+    public Builder tables(List<Table> tables) {
+      this.tables = tables;
+      return this;
+    }
+
+    public Builder where(FilterVariable where) {
+      this.where = where;
+      return this;
+    }
+
+    public Builder groupBy(List<FieldVariable> groupBy) {
+      this.groupBy = groupBy;
+      return this;
+    }
+
+    public Builder orderBy(List<OrderByVariable> orderBy) {
+      this.orderBy = orderBy;
+      return this;
+    }
+
+    public Builder limit(Integer limit) {
+      this.limit = limit;
+      return this;
+    }
+
+    public Query build() {
+      return new Query(
+          this.select, this.tables, this.where, this.groupBy, this.orderBy, this.limit);
```

Comment:
In general, the TDR code only uses `this.` in places where it's required for disambiguation purposes, like a constructor or setter. So to use the TDR convention:
```suggestion
          select, tables, where, groupBy, orderBy, limit);
```